mkosi/__init__.py

```diff
@@ -3296,7 +3296,8 @@ def need_cache_tree(state: MkosiState) -> bool:
 
 
 def build_stuff(uid: int, gid: int, config: MkosiConfig) -> None:
-    workspace = tempfile.TemporaryDirectory(dir=config.workspace_dir or Path.cwd(), prefix=".mkosi.tmp")
+    make_output_dir(config, uid, gid)
+    workspace = tempfile.TemporaryDirectory(dir=config.workspace_dir or config.output_dir or Path.cwd(), prefix=".mkosi.tmp")
```

Comment:
Maybe the fallback to the current working directory should also be explicitly mentioned in the docs.

Reply:
The current working directory should be the output directory if it is not specified so it is there implicitely, but I can add a comment along the lines of 
> ..., which is the current working directory if not specified otherwise with the config option `OutputDirectory=` or the commandline flags `--output-dir=` or  `-O`...

Reply:
Yes, it's implicitly clear, but explicit is better than implicit, since it requires less mental load on the reader to make the connection.